compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1389,11 +1389,16 @@ getPackVectorTileSizes(mlir::FunctionOpInterface entryPointFn,
   SmallVector<int64_t> tileSizes(op.getSourceRank(), 1);
   auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
   int64_t vectorSize = getVectorSize(entryPointFn, op.getSourceType());
+  if (!(hasAVX512fFeature(targetAttr) && isPackMatmulLHS(op))) {
+    return tileSizes;
+  }
   // TODO(#15421): Improve tile sizes selection for non f32 cases.
```

Comment:
Since you're touching the code, can you help update the TODO ID to `16314`? I think we now use https://github.com/iree-org/iree/issues/16314 to track the issues.